src/Clearfix.js

```diff
@@ -1,71 +1,79 @@
-import React from 'react';
 import classNames from 'classnames';
-import styleMaps from './styleMaps';
+import React from 'react';
 import elementType from 'react-prop-types/lib/elementType';
 
-const Clearfix = React.createClass({
-  propTypes: {
-    /**
-     * You can use a custom element for this component
-     */
-    componentClass: elementType,
-    /**
-     * Apply clearfix
-     *
-     * on Extra small devices Phones
-     *
-     * adds class `visible-xs-block`
-     */
-    visibleXsBlock: React.PropTypes.bool,
-    /**
-     * Apply clearfix
-     *
-     * on Small devices Tablets
-     *
-     * adds class `visible-sm-block`
-     */
-    visibleSmBlock: React.PropTypes.bool,
-    /**
-     * Apply clearfix
-     *
-     * on Medium devices Desktops
-     *
-     * adds class `visible-md-block`
-     */
-    visibleMdBlock: React.PropTypes.bool,
-    /**
-     * Apply clearfix
-     *
-     * on Large devices Desktops
-     *
-     * adds class `visible-lg-block`
-     */
-    visibleLgBlock: React.PropTypes.bool
-  },
+import { bsClass, getClassSet, omitBsProps } from './utils/bootstrapUtils';
+import capitalize from './utils/capitalize';
+import { DEVICE_SIZES } from './utils/StyleConfig';
+
+const propTypes = {
+  componentClass: elementType,
 
-  getDefaultProps() {
-    return {
-      componentClass: 'div'
-    };
-  },
+  // TODO: These should be `visibleSz`, not `visibleSzBlock`.
```

Comment:
should they be? the classes without display stlye are deprecated. If anything we should allow `inline` and `inline-block` yeah?